src/torchmetrics/wrappers/transformations.py

```diff
@@ -0,0 +1,175 @@
+    forwarding all calls to the wrapped metric with modifications applied.
+
+    """
+
+    def __init__(self, wrapped_metric: Union[Metric, MetricCollection], **kwargs: Dict[str, Any]) -> None:
+        super().__init__(**kwargs)
+        if not isinstance(wrapped_metric, (Metric, MetricCollection)):
+            raise TypeError(
+                f"Expected wrapped metric to be an instance of `torchmetrics.Metric` or "
+                f"`torchmetrics.MetricsCollection`but received {wrapped_metric}"
+            )
+        self.wrapped_metric = wrapped_metric
+
+    def transform_pred(self, pred: torch.Tensor) -> torch.Tensor:
+        """Define transform operations on the prediction data.
+
+        Overridden by subclasses. Identity by default.
+
+        """
+        return pred
+
+    def transform_target(self, target: torch.Tensor) -> torch.Tensor:
+        """Define transform operations on the target data.
+
+        Overridden by subclasses. Identity by default.
+
+        """
+        return target
+
+    def _wrap_transform(self, *args: torch.Tensor) -> Tuple[torch.Tensor, ...]:
+        """Wrap transformation functions to dispatch args to their individual transform functions."""
+        if len(args) == 1:
+            return (self.transform_pred(args[0]),)
+        if len(args) == 2:
+            return self.transform_pred(args[0]), self.transform_target(args[1])
+        return self.transform_pred(args[0]), self.transform_target(args[1]), *args[2:]
+
+    def update(self, *args: torch.Tensor, **kwargs: Dict[str, Any]) -> None:
+        """Wrap the update call of the underlying metric."""
+        args = self._wrap_transform(*args)
+        self.wrapped_metric.update(*args, **kwargs)
+
+    def compute(self) -> Any:
+        """Wrap the compute call of the underlying metric."""
+        return self.wrapped_metric.compute()
+
+    def forward(self, *args: torch.Tensor, **kwargs: Dict[str, Any]) -> Any:
+        """Wrap the forward call of the underlying metric."""
+        args = self._wrap_transform(*args)
+        return self.wrapped_metric.forward(*args, **kwargs)
+
+
+class LambdaInputTransformer(MetricInputTransformer):
+    """Wrapper class for transforming a metrics' inputs given a user-defined lambda function.
+
+    Args:
+        wrapped_metric:
+            The underlying `Metric` or `MetricCollection`.
+        transform_pred:
+            The function to apply to the predictions before computing the metric.
+        transform_target:
+            The function to apply to the target before computing the metric.
+
+    Raises:
+        TypeError:
+            If `transform_pred` is not a Callable.
+        TypeError:
+            If `transform_target` is not a Callable.
+
+    Example:
+        >>> import torch
+        >>> from torchmetrics.wrappers import LambdaInputTransformer
+        >>> from torchmetrics.classification import BinaryAccuracy
```

Comment:
Complete nitpick, I just see `isort/ruff` everywhere :D 

```suggestion
        >>> from torchmetrics.classification import BinaryAccuracy
        >>> from torchmetrics.wrappers import LambdaInputTransformer
```